File: JavaScript/0125_valid_palindrome/solution.js
function valid_palindrome(string) {
    let result = "";
    for (let char of string) {
        if ((char >= 'a' && char <= 'z') || (char >= 'A' && char <= 'Z') || (char >= '0' && char <= '9')) {
            result += char;
        }
    }
    const reversed_string = result.split("").reverse().join("");
    if (result.toLowerCase() == reversed_string.toLowerCase()) {
        return true;
    }
    return false;
}

const string = "A man, a plan, a canal: Panama";
console.log(valid_palindrome(string));

File: Python/0125_valid_palindrome/solution.py
def valid_palindrome(string):
    result = ""
    for char in string:
        if char.isalnum():
            result += char
    
    reversed_string = result[::-1]
    if result.lower() == reversed_string.lower():
        return True
    return False

string = "A man, a plan, a canal: Panama"
print(valid_palindrome(string))